### DISCUSSION TITLE
Missing Plugs (Not a bug)

### DISCUSSION THREAD

**Discussion by dimaguy** [CosmosOS/Cosmos#1605] [Q&A]
I have been trying to get Moonsharp lib to work with Cosmos, but it is reporting as if system plugs are missing.  

`Native code encountered, plug required.`  
`1>  DO NOT REPORT THIS AS A BUG.`
`1>  Please see http://www.gocosmos.org/docs/plugs/missing/`
`1>  Need plug for: System.String  System.Number.FormatDouble(System.Double, System.String, System.Globalization.NumberFormatInfo).`
`1>  Called from :`
`1>System.Double::System.String ToString(System.IFormatProvider)`
`1>MoonSharp.Interpreter.DynValue::System.String ToString()`
`1>MoonSharp.Interpreter.DynValue`  

Is there something I can do to make it possible (besides writing a plug since I have 0 knowledge in asm)?

**Comment by zarlo** [comment]
to make some plugs alot of the times it can be done in 100% c# others will need some x#
but Moonsharp it will need a lot of plug form memory it uses reflection in it and there is still abit of work that is needed to get the working
